analyzers/src/SonarAnalyzer.CSharp/Rules/NewGuidShouldNotBeUsed.cs

```diff
@@ -33,27 +35,56 @@ namespace SonarAnalyzer.Rules.CSharp
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            DetectIssueInConstructors(context);
+            DetectIssueInDefaultExpressions(context);
+        }
+
+        private static void DetectIssueInConstructors(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var objectCreationSyntax = ObjectCreationFactory.Create(c.Node);
+                var type = objectCreationSyntax.MethodSymbol(c.SemanticModel);
+                if (objectCreationSyntax.ArgumentList?.Arguments.Count == 0
+                    && type != null
+                    && type.ContainingType.Is(KnownType.System_Guid))
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, objectCreationSyntax.Expression.GetLocation()));
+                }
+            },
+            SyntaxKind.ObjectCreationExpression,
+            SyntaxKindEx.ImplicitObjectCreationExpression);
+
+        private static void DetectIssueInDefaultExpressions(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var expressionSyntax = (ExpressionSyntax)c.Node;
+                if (expressionSyntax.IsKind(SyntaxKindEx.DefaultLiteralExpression)
+                    && c.SemanticModel.GetTypeInfo(expressionSyntax).ConvertedType.Is(KnownType.System_Guid))
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, expressionSyntax.GetLocation()));
+                }
+                else if (expressionSyntax.IsKind(SyntaxKind.DefaultExpression)
+                         && DefaultExpressionIdentifierIsGuid((DefaultExpressionSyntax)expressionSyntax))
                 {
-                    var objectCreationSyntax = (ObjectCreationExpressionSyntax)c.Node;
+                    c.ReportIssue(Diagnostic.Create(Rule, expressionSyntax.GetLocation()));
+                }
+            },
+            SyntaxKind.DefaultExpression,
+            SyntaxKindEx.DefaultLiteralExpression);
 
-                    if (objectCreationSyntax.ArgumentList?.Arguments.Count == 0 &&
-                        c.SemanticModel.GetSymbolInfo(objectCreationSyntax).Symbol is IMethodSymbol methodSymbol &&
-                        methodSymbol.ContainingType.Is(KnownType.System_Guid))
-                    {
-                        c.ReportIssue(Diagnostic.Create(rule, objectCreationSyntax.GetLocation()));
-                    }
-                },
-                SyntaxKind.ObjectCreationExpression);
+        private static bool DefaultExpressionIdentifierIsGuid(DefaultExpressionSyntax defaultExpression)
+        {
+            var expressionIdentifier = defaultExpression.Type.GetIdentifier();
+            return expressionIdentifier != null
+                ? expressionIdentifier.Identifier.Text.Equals("Guid")
+                : defaultExpression.Type.ToString().Equals("System.Guid");
```

Comment:
I think we can simplify this a bit:
```suggestion
            var typeName = defaultExpression.Type.ToString();
            return typeName == "Guid" || typeName == "System.Guid";
```
It will not work with aliases but I think that's good enough for now.

Reply:
Yes and it's a false negative, so it won't create too much noise.